src/tasks/leveling.ts

```diff
@@ -262,13 +312,34 @@ export const LevelingQuest: Quest = {
     {
       name: "Clan Shower",
       completed: () => get("_aprilShower"),
-      do: () => ensureEffect($effect`Thaumodynamic`),
+      do: (): void => {
+        const aprilShowerEffect: Effect = {
+          Muscle: $effect`Muscle Unbound`,
+          Mysticality: $effect`Thaumodynamic`,
+          Moxie: $effect`So Fresh and So Clean`,
+        }[mainStatStr];
+        ensureEffect(aprilShowerEffect);
+      },
       limit: { tries: 1 },
     },
     {
-      name: "Inscrutable Gaze",
-      completed: () => have($effect`Inscrutable Gaze`) || !have($skill`Inscrutable Gaze`),
-      do: (): void => ensureEffect($effect`Inscrutable Gaze`),
+      name: "Mainstat Gaze",
+      completed: () =>
+        ((have($effect`Inscrutable Gaze`) || !have($skill`Inscrutable Gaze`)) &&
+          mainStat === $stat`Mysticality`) ||
+        ((have($effect`Patient Smile`) || !have($skill`Patient Smile`)) &&
+          mainStat === $stat`Muscle`) ||
+        ((have($effect`Knowing Smile`) || !have($skill`Knowing Smile`)) &&
+          mainStat === $stat`Moxie`),
+      do: (): void => {
+        const mainStatGainEffect: Effect = {
+          Muscle: $effect`Patient Smile`,
+          Mysticality: $effect`Inscrutable Gaze`,
+          Moxie: $effect`Knowing Smile`,
+        }[mainStatStr];
+        ensureEffect(mainStatGainEffect);
+      },
+      limit: { tries: 1 },
```

Comment:
This limit needs to be raised because we do use other gazes (e.g. Intimidating Mien) for certain tasks, and might want to swap back after.

5 seems to be a reasonable number.

Reply:
This doesn't seem to be resolved yet.